### PR TITLE
upgrade lm-eval to 0.4.2

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

feature:
upgrade lm-eval, accuracy will fluctuation a little due to seed change, please aware. (seed from 1234,1234,1234 to 0,1234,1234)
support chinese dataset evaluation.

please assign resource to modify if INC used lm-eval. @thuang6 
please assign resource to modify if NS used lm-eval. @a32543254 
please assign resource to check and review NeuralChat use lm-eval correct, I already change the code. @lvliang-intel 
please review onnx part and please update if it is used by INC or ITREX examples. @mengniwang95 

## Description

detail description 
update lm-eval to the latest release version 0.4.2
**usage:**
### torch
```shell
#causal
from intel_extension_for_transformers.transformers.llm.evaluation.lm_eval import evaluate, LMevalParser
args = LMevalParser(model = "hf", 
                    model_args="pretrained=hf-internal-testing/tiny-random-t5,dtype=float32",
                    tasks = "piqa",
                    device = "cpu",
                    batch_size = 1,
                    limit = 5)
results = evaluate(args)
# seq2seqLM
from intel_extension_for_transformers.transformers.llm.evaluation.lm_eval import evaluate, LMevalParser
args = LMevalParser(model = "hf", 
                    model_args="pretrained=./t5-past,dtype=float32,model_format=onnx",
                    tasks = "piqa",
                    device = "cpu",
                    batch_size = 1,
                    limit = 5)
results = evaluate(args)
```
### onnx
```
from intel_extension_for_transformers.transformers.llm.evaluation.lm_eval import evaluate, LMevalParser
args = LMevalParser(model = "hf", 
                    model_args="pretrained=./gptj-past,dtype=float32,model_format=onnx",
                    tasks = "piqa",
                    device = "cpu",
                    batch_size = 1,
                    limit = 5)
results = evaluate(args)
```
### neural-speed
```
from intel_extension_for_transformers.transformers.llm.evaluation.lm_eval import evaluate, LMevalParser
args = LMevalParser(model = "hf", 
                    model_args="pretrained=TheBloke/Llama-2-7B-Chat-GPTQ,model_format=neural_speed",
                    tasks = "piqa",
                    device = "cpu",
                    batch_size = 1,
                    limit = 5)
results = evaluate(args)
```
### smoothquant w8a8 model
```
from intel_extension_for_transformers.transformers.llm.evaluation.lm_eval import evaluate, LMevalParser
args = LMevalParser(model = "hf", 
                    user_model = user_model,
                    tokenizer = tokenizer,
                    tasks = "piqa",
                    device = "cpu",
                    batch_size = 1,
                    limit = 5)
results = evaluate(args)
```

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by thuang6** [comment]
@changwangss  Thanks for reminding, Xin will look at INC side and raise PR to upgrade for examples.

File: examples/huggingface/pytorch/language-modeling/inference/run_clm_no_trainer.py [hunk 1]
```diff
@@ -83,23 +82,21 @@
...
         )
-        dumped = json.dumps(results, indent=2)
-        if args.save_accuracy_path:
-            with open(args.save_accuracy_path, "w") as f:
-                f.write(dumped)
-        for task_name in args.tasks:
+        results = evaluate(eval_args)
+        for task_name in args.tasks.split(","):
             if task_name == "wikitext":
                 print("Accuracy for %s is: %s" % (task_name, results["results"][task_name]["word_perplexity"]))
```
Comment (xin3he) [inline examples/huggingface/pytorch/language-modeling/inference/run_clm_no_trainer.py:97]:
```suggestion
                print("Accuracy for %s is: %s" % (task_name, results["results"][task_name]["word_perplexity,none"]))

```

**Review by xin3he — APPROVED** [review]

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/job/nlp-toolkit-validation-top-mr-extension/1512/

**Review by mengniwang95 — APPROVED** [review]

File: examples/huggingface/pytorch/code-generation/quantization/requirements.txt [hunk 2]
```diff
@@ -12,4 +12,4 @@ neural-compressor
 intel_extension_for_pytorch==2.2.0
 optimum-intel
 auto-round
-git+https://github.com/bigcode-project/bigcode-evaluation-harness@00967d12093ef614de7bdad0772aed8e4118f1fd
+git+https://github.com/bigcode-project/bigcode-evaluation-harness@094c7cc197d13a53c19303865e2056f1c7488ac1
```
Comment (a32543254) [inline examples/huggingface/pytorch/code-generation/quantization/requirements.txt:15]: why for quantization we still need commit id  and  for inference we don't need it

File: examples/huggingface/pytorch/text-generation/quantization/requirements_cpu_woq.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+protobuf
+sentencepiece != 0.1.92
+--extra-index-url https://download.pytorch.org/whl/cpu
+torch==2.2.0+cpu
+transformers
+bitsandbytes  #baichuan
+transformers_stream_generator
+tiktoken  #qwen
+einops  #qwen
+git+https://github.com/intel/neural-speed.git@v1.0.1.dev0
```
Comment (a32543254) [inline examples/huggingface/pytorch/text-generation/quantization/requirements_cpu_woq.txt:13]: do we need the version for ns? 
I think we can follow the inc just 
git+https://github.com/intel/neural-speed.git

File: examples/huggingface/pytorch/code-generation/quantization/requirements.txt (hunk 2 above)
Comment (changwangss) [reply examples/huggingface/pytorch/code-generation/quantization/requirements.txt:15]: it is bigcode-evaluation-harness, not lm-eval and used only for code-generation, it doesn't have pip release version, only support source install.
https://github.com/bigcode-project/bigcode-evaluation-harness

File: examples/huggingface/pytorch/text-generation/quantization/requirements_cpu_woq.txt (hunk 3 above)
Comment (changwangss) [reply examples/huggingface/pytorch/text-generation/quantization/requirements_cpu_woq.txt:13]: improved.

File: examples/huggingface/neural_speed/run_accuracy.py [hunk 4]
```diff
@@ -25,22 +25,22 @@
     args = parser.parse_args()
     print(args)
     model_args=f'pretrained="{args.model_name}",dtype=float32,trust_remote_code=True'
-    if args.use_gptq:
-        model_args += ",use_gptq=True"
```
Comment (a32543254) [inline examples/huggingface/neural_speed/run_accuracy.py:29]: seems one args removed here~ 
no problem right ? @zhentaoyu

Comment (zhentaoyu) [reply examples/huggingface/neural_speed/run_accuracy.py:29]: no effect on `NS` since it will be handled by model_config.

**Review by a32543254 — APPROVED** [review]
LGTM

File: examples/huggingface/neural_speed/run_accuracy.py [hunk 5]
```diff
@@ -25,22 +25,22 @@
...
+
     if args.model_format == "neural_speed":
-        results = evaluate(
-            model="hf-causal",
-            model_args=model_args,
-            batch_size=args.batch_size,
-            tasks=[f"{args.tasks}"],
-            model_format=f"{args.model_format}"
-        )
+        eval_args = LMEvalParser(model="hf",
```
Comment (intellinjun) [inline examples/huggingface/neural_speed/run_accuracy.py:30]: It doesn't go inside neural_speed

**Comment by changwangss** [comment]
extension.
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1525/artifact/report.html
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/nlp-toolkit-validation-top-mr-extension/1526/

File: examples/huggingface/neural_speed/run_accuracy.py (hunk 5 above)
Comment (changwangss) [reply examples/huggingface/neural_speed/run_accuracy.py:30]: here is referenece, please make it run successfully, please call me if something needed. :)
```
    model_args=f'pretrained="{args.model_name}",dtype=float32,trust_remote_code=True'

    if args.model_format == "neural_speed":
        model_args += f",model_format=neural_speed"
```

**Comment by VincyZhang** [comment]
model test: https://inteltf-jenk.sh.intel.com/job/nlp-toolkit-weekly/91/

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/job/nlp-toolkit-weekly/92/

**Comment by changwangss** [comment]
[nlp_toolkit_optimize_validation_localtest #6326 [Jenkins] (intel.com)](https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/6326/)
[nlp_toolkit_optimize_validation_localtest #6327 [Jenkins] (intel.com)](https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/6327/)

**Comment by changwangss** [comment]
chatglm2 https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/6347/

**Comment by changwangss** [comment]
[nlp_toolkit_optimize_validation_localtest #6348 Console [Jenkins] (intel.com)](https://inteltf-jenk.sh.intel.com/job/nlp_toolkit_optimize_validation_localtest/6348/console)

**Pull request merged**